pkgs/tools/system/collectd/plugins.nix

```
@@ -46,299 +49,90 @@ let
-    ovs_events = {
-      buildInputs = [ yajl ];
-    };
-    ovs_stats = {
-      buildInputs = [ yajl ];
-    };
-    perl = {
-      buildInputs = [ perl ];
-    };
-    pf = {};
-    pinba = {
-      buildInputs = [ protobufc ];
-    };
-    ping = {
-      buildInputs = [ liboping ];
-    };
-    postgresql = {
-      buildInputs = [ postgresql ];
-    };
-    powerdns = {};
-    processes = {};
-    protocols = {};
-    python = {
-      buildInputs = [ python ];
-    };
-    redis = {
-      buildInputs = [ hiredis ];
-    };
-    routeros = {};
-    rrdcached = {
-      buildInputs = [ rrdtool libxml2 ];
-    };
-    rrdtool = {
-      buildInputs = [ rrdtool libxml2 ];
-    };
-    sensors = {
-      buildInputs = lib.optionals stdenv.isLinux [ lm_sensors ];
-    };
-    serial = {};
-    sigrok = {
-      buildInputs = lib.optionals stdenv.isLinux [ libsigrok udev ];
-    };
-    smart = {
-      buildInputs = lib.optionals stdenv.isLinux [ libatasmart udev ];
-    };
-    snmp = {
-      buildInputs = lib.optionals stdenv.isLinux [ net-snmp ];
-    };
-    snmp_agent = {
-      buildInputs = lib.optionals stdenv.isLinux [ net-snmp ];
-    };
-    statsd = {};
-    swap = {};
-    synproxy = {};
-    syslog = {};
-    table = {};
-    tail_csv = {};
-    tail = {};
-    tape = {};
-    target_notification = {};
-    target_replace = {};
-    target_scale = {};
-    target_set = {};
-    target_v5upgrade = {};
-    tcpconns = {};
-    teamspeak2 = {};
-    ted = {};
-    thermal = {};
-    threshold = {};
-    tokyotyrant = {};
-    turbostat = {};
-    unixsock = {};
-    uptime = {};
-    users = {};
-    uuid = {};
-    varnish = {
-      buildInputs = [ curl varnish ];
-    };
-    virt = {
-      buildInputs = [ libvirt libxml2 yajl ] ++
-        lib.optionals stdenv.isLinux [ lvm2 udev
-          # those might be no longer required when https://github.com/NixOS/nixpkgs/pull/51767
-          # is merged
-          libapparmor numactl libcap_ng
-        ];
-    };
-    vmem = {};
-    vserver = {};
-    wireless = {};
-    write_graphite = {};
-    write_http = {
-      buildInputs = [ curl yajl ];
-    };
-    write_kafka = {
-      buildInputs = [ yajl rdkafka ];
-    };
-    write_log = {
-      buildInputs = [ yajl ];
-    };
-    write_mongodb = {
-      buildInputs = [ mongoc ];
-    };
-    write_prometheus = {
-      buildInputs = [ protobufc libmicrohttpd ];
-    };
-    write_redis = {
-      buildInputs = [ hiredis ];
-    };
-    write_riemann = {
-      buildInputs = [ protobufc riemann_c_client ];
-    };
-    write_sensu = {};
-    write_tsdb = {};
-    xencpu = {};
-    xmms = {};
-    zfs_arc = {};
-    zone = {};
-    zookeeper = {};
+    amqp.buildInputs = [
+      yajl
+    ] ++ lib.optionals stdenv.isLinux [ rabbitmq-c ];
+    apache.buildInputs = [ curl ];
+    ascent.buildInputs = [ curl libxml2 ];
+    battery.buildInputs = lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.IOKit
```

```suggestion
      IOKit
```
Please do ``imherit (darwin.apple_sdk.frameworks) IOKit`` in the top level call.